packages/api/src/controllers/asset.ts

```diff
@@ -418,7 +418,23 @@ function getDownloadUrl(
     os.id !== vodObjectStoreId ? os.publicUrl : pathJoin(ingest, "asset");
   const source = asset.files?.find((f) => f.type === "source_file");
   if (source) {
-    return pathJoin(base, asset.playbackId, source.path);
+    const formatToFileExtension: Map<string, string> = new Map([
+      ["mpegts", "ts"],
+      ["mp4", "mp4"],
+    ]);
+
+    if (
+      source.path != "video" ||
+      !formatToFileExtension.has(asset.videoSpec.format)
+    ) {
+      return pathJoin(base, asset.playbackId, source.path);
+    }
+    return pathJoin(
+      base,
+      asset.playbackId,
+      source.path,
+      "download." + formatToFileExtension.get(asset.videoSpec.format),
```

Comment:
wdyt of naming the file with the asset name instead?
```suggestion
      encodeURIComponent(`${asset.name}.${formatToFileExtension.get(asset.videoSpec.format)}`),
```

Reply:
Only issue with that is that the name for direct uploads will contain the original file extension and so they'll end up with double file extensions :(
<img width="379" alt="Screenshot 2024-09-19 at 12 21 13" src="https://github.com/user-attachments/assets/87355450-fecf-4f8c-8a7b-477c531acc37">

Reply:
could just strip anything after the final `.`?